### PR TITLE
Update substrate to v0.9.28

### PR DESCRIPTION
**Pull request by Yithis**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have made corresponding changes to the existing documentation
- I have created new documentation
- I have bumped `spec_version` and `transaction_version`
- I have bumped aleph-client version if relevant

### CODE REVIEW

File: e2e-tests/Cargo.toml [hunk 1]
```diff
@@ -14,12 +14,15 @@ serde_json = "1.0"
 codec = { package = 'parity-scale-codec', version = "3.0", default-features = false, features = ['derive'] }
 rayon = "1.5"
 
-sp-core = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", default-features = false, features = ["full_crypto"] }
-sp-runtime = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", default-features = false }
-frame-support = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", default-features = false }
-system = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", package = "frame-system" }
-pallet-staking = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", default-features = false }
-pallet-balances = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.26", default-features = false }
+substrate-api-client = { git = "https://github.com/Cardinal-Cryptography/substrate-api-client.git", branch = "aleph-v0.9.28", features = ["staking-xt"] }
```
Comment (fixxxedpoint) [inline e2e-tests/Cargo.toml:17]: Can you remove direct dependency on `substrate-api-client` in e2e-tests?

Comment (Yithis) [reply e2e-tests/Cargo.toml]: Done

**Review by mike1729 — APPROVED** [review]
I approve under an assumption that this PR is not going to be merged :P

File: bin/node/src/main.rs [hunk 2]
```diff
@@ -15,11 +15,11 @@ fn main() -> sc_cli::Result<()> {
         .run
         .import_params
         .pruning_params
-        .pruning
-        .replace(String::from("archive"))
-        .map_or(false, |x| x != "archive")
+        .blocks_pruning
```
Comment (DamianStraszak) [inline bin/node/src/main.rs:18]: Is there both `blocks_pruning` and `state_pruning` in the config now? 

If so, then this change is not quite equivalent to what was there before  -- I think you just want to disable both types of pruning.

Btw. I think we should have here a little "switch" in the form of a cli argument like `--experimental-pruning` which does not enforce pruning. This way we can test pruning without recompiling over and over again.

File: bin/runtime/src/lib.rs [hunk 3]
```diff
@@ -107,7 +107,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("aleph-node"),
     impl_name: create_runtime_str!("aleph-node"),
     authoring_version: 1,
-    spec_version: 34,
+    spec_version: 35,
     impl_version: 1,
     apis: RUNTIME_API_VERSIONS,
     transaction_version: 9,
```
Comment (DamianStraszak) [inline bin/runtime/src/lib.rs:113]: bump this as well.

File: bin/node/src/main.rs (hunk 2 above)
Comment (DamianStraszak) [reply bin/node/src/main.rs:18]: The "switch" does not belong to this PR though (unless it's trivial).

File: bin/runtime/src/lib.rs (hunk 3 above)
Comment (Yithis) [reply bin/runtime/src/lib.rs]: Done

File: bin/node/src/main.rs (hunk 2 above)
Comment (Yithis) [reply bin/node/src/main.rs:18]: Done as discussed yesterday

**Review by DamianStraszak — APPROVED** [review]

**Pull request merged**